src/brod.erl

```diff
@@ -89,27 +89,144 @@
 
 -deprecated([{start_link_client, '_', next_version}]).
 
--export_type([ brod_call_ref/0
-             , brod_client_id/0
-             , brod_partition_fun/0
-             , client/0
+-include("brod.hrl").
+-include("brod_int.hrl").
+
+%%%_* Types ====================================================================
+-type hostname() :: string().
+-type portnum()  :: pos_integer().
+-type endpoint() :: {hostname(), portnum()}.
+
+-export_type([ hostname/0
+             , portnum/0
+             , endpoint/0
+             ]).
+
+-type consumer_option() :: begin_offset
+                         | min_bytes
+                         | max_bytes
+                         | max_wait_time
+                         | sleep_timeout
+                         | prefetch_count.
+-type consumer_options() :: [{consumer_option(), integer()}].
+
+-export_type([ consumer_options/0
+             ]).
+
+
+-type leader_id()       :: non_neg_integer().
+-type client_config()   :: brod_client_config().
+-type producer_config() :: brod_producer_config().
+-type consumer_config() :: brod_consumer_config().
+-type client()          :: brod_client_id() | pid().
+-type required_acks()   :: -1..1.
+
+-export_type([ leader_id/0
```

Comment:
They were grouped in the same way in the header files. It seemed logical to keep the grouping in brod.erl as well. As a consequence it made more sense to me to export them as such groups rather than one big export_type block. 

If you disagree I'm fine with building one export_type block. My intention was to keep as much of the original grouping and separation as possible :)